### PR TITLE
Improves exceptions logging

### PR DESCRIPTION
**Pull request by BertrandBordage**
While using [ui/django-rq](https://github.com/ui/django-rq) and `django.utils.log.AdminEmailHandler`, I faced two issues:
- `UnpickleError` wasn't caught, and therefore I didn't get any mail when incorrect arguments were passed to RQ;
- Mails were poorly formatted.

At first I thought it was coming from django-rq, but after digging I realized this was coming from RQ.  This pull request fixes these two issues.

### CODE REVIEW

File: rq/worker.py [hunk 1]
```diff
@@ -474,8 +474,17 @@ def perform_job(self, job):
         self.set_current_job_id(job.id)
         self.heartbeat((job.timeout or 180) + 60)
 
+        try:
+            func_name = job.func_name
+        except UnpickleError:
+            # Use the public setter here, to immediately update Redis
+            job.set_status(Status.FAILED)
+            self.log.exception('RQ failed to unpickle job %s' % job.id)
```
Comment (chaffeqa) [inline rq/worker.py]: In dealing with sentry, I've found the best solution is to do simple things like:

```python
except Exception as e:
    self.log.exception(e, extra={...})
```

With any got exception handler, passing the exception itself is the most supported approach, and sentry especially does a good job giving context and insight.

Basically you can reduce a ton of code and verbosity, as well as improve support.

Comment (chaffeqa) [inline rq/worker.py]: Which leads into my next proposal:  instead of passing `exc_info` to handlers, I'd rather get the `Exception` instance

**Comment by untitaker** [comment]
FYI https://github.com/rq/rq/issues/1003 is probably relevant: sentry-sdk on PyPI has a new RQ integration that Sentry officially supports

**Comment by selwin** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**